pkg/gardenlet/controller/seed/seed_care_control.go

```diff
@@ -0,0 +1,214 @@
+func NewCareReconciler(
+	clientMap clientmap.ClientMap,
+	seedCareControllerConfig gardenletconfig.SeedCareControllerConfiguration,
+) reconcile.Reconciler {
+	return &careReconciler{
+		clientMap:                clientMap,
+		seedCareControllerConfig: seedCareControllerConfig,
+	}
+}
+
+type careReconciler struct {
+	clientMap                clientmap.ClientMap
+	seedCareControllerConfig gardenletconfig.SeedCareControllerConfiguration
+}
+
+func (r *careReconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	log := logf.FromContext(ctx).WithName(req.Name)
+
+	gardenClient, err := r.clientMap.GetClient(ctx, keys.ForGarden())
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to get garden client: %w", err)
+	}
+
+	seed := &gardencorev1beta1.Seed{}
+	if err := gardenClient.Client().Get(ctx, req.NamespacedName, seed); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("Stopping care operations for Seed since it has been deleted")
+			return reconcile.Result{}, nil
+		}
+		log.Error(err, "Unable to retrieve object from store")
+		return reconcile.Result{}, err
+	}
+
+	if err := r.care(ctx, gardenClient.Client(), seed, log); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{RequeueAfter: r.seedCareControllerConfig.SyncPeriod.Duration}, nil
+}
+
+var (
+	// NewSeed is used to create a new `operation.Operation` instance.
+	NewSeed = defaultNewSeedObjectFunc
+	// NewHealthCheck is used to create a new Health check instance.
+	NewHealthCheck = defaultNewHealthCheck
+)
+
+func (r *careReconciler) care(ctx context.Context, gardenClientSet client.Client, seed *gardencorev1beta1.Seed, log logr.Logger) error {
+	careCtx, cancel := context.WithTimeout(ctx, r.seedCareControllerConfig.SyncPeriod.Duration)
+	defer cancel()
+
+	log.V(1).Info("Starting seed care")
+
+	// Initialize conditions based on the current status.
+	conditionTypes := []gardencorev1beta1.ConditionType{
+		gardencorev1beta1.SeedSystemComponentsHealthy,
+	}
+	var conditions []gardencorev1beta1.Condition
+	for _, cond := range conditionTypes {
+		conditions = append(conditions, gardencorev1beta1helper.GetOrInitCondition(seed.Status.Conditions, cond))
+	}
+
+	seedClient, err := r.clientMap.GetClient(careCtx, keys.ForSeed(seed))
+	if err != nil {
+		log.Error(err, "SeedClient cannot be constructed")
+
+		if err := careSetupFailure(ctx, gardenClientSet, seed, "Precondition failed: seed client cannot be constructed", conditions); err != nil {
+			log.Error(err, "Unable to create error condition")
+		}
+
+		return nil // We do not want to run in the exponential backoff for the condition checks.
+	}
+
+	seedObj, err := NewSeed(careCtx, seed)
+	if err != nil {
+		log.Error(err, "SeedObj cannot be constructed")
+		if err := careSetupFailure(careCtx, gardenClientSet, seed, "seedObj cannot be constructed", conditions); err != nil {
```

Comment:
Why is a different context used here than in line 104?
Can you always use `ctx` for update calls to the shoot resource?